src/components/frame/FrameConnector.tsx

```diff
@@ -46,50 +35,44 @@ interface LegacyFrameConnectorProps extends BaseFrameConnectorProps {
  * - a `dispatch `function that will listen for actions coming from the frame
  * - the URL of the decentralized renderer to use as the `source` prop
  */
-export const FrameConnector: FunctionComponent<FrameConnectorProps | LegacyFrameConnectorProps> = ({
+export const FrameConnector: FunctionComponent<FrameConnectorProps> = ({
   dispatch,
-  methods,
   source,
   onConnected,
   style,
   className = ""
 }) => {
   const iframe = useRef<HTMLIFrameElement>(null);
 
-  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-  // @ts-ignore error because of typescript distributivity
-  // FrameConnectorProps can me mapped to UseLegacyChildrenFrameProps which is not correct. It could be solved using
-  // conditional but there is a hooks rules saying shouldn't use conditional
+  const methods = useMemo<LegacyFrameActions>(() => {
+    return {
+      updateHeight: (height: number) => dispatch(updateHeight(height)),
+      updateTemplates: templates => dispatch(updateTemplates(templates)),
+      handleObfuscation: field => dispatch(obfuscateField(field))
+    };
+  }, [dispatch]);
   const [connected, toFrame] = useChildFrame({ methods, dispatch, iframe });
   useEffect(() => {
     if (connected) {
       onConnected(
-        Object.assign(
-          (action: HostActions) => {
-            if (toFrame.dispatch) {
-              toFrame.dispatch(action);
-            }
-          },
-          {
-            renderDocument: (document: Document, rawDocument?: WrappedDocument<Document>) => {
-              if (toFrame.renderDocument) {
-                toFrame.renderDocument(document, rawDocument);
-              }
-            },
-            selectTemplateTab: (tabIndex: number) => {
-              if (toFrame.selectTemplateTab) {
-                toFrame.selectTemplateTab(tabIndex);
-              }
-            },
-            print: () => {
-              if (toFrame.print) {
-                toFrame.print();
-                return true;
-              }
-              return false;
+        Object.assign((action: HostActions) => {
+          if (toFrame.dispatch) {
+            toFrame.dispatch(action);
+          } else if (action.type === "RENDER_DOCUMENT" && toFrame.renderDocument) {
```

Comment:
```suggestion
          if (dispatchExist) {
            toFrame.dispatch(action);
          } else {
            handleLegacyMethods(...);
          }
```

Thinking of something like this to better explain the difference

Reply:
I followed that pattern if/else but I kept the logic within the else block (no indirection to another function) + I added comments